vms/proposervm/vm_test.go

```diff
@@ -810,6 +809,18 @@ func TestExpiredBuildBlock(t *testing.T) {
 		}
 	}
 
+	testEnd := make(chan struct{})
+
+	events := make(chan common.Message, 1)
+	coreVM.WaitForEventF = func(context.Context) (common.Message, error) {
+		select {
+		case <-testEnd:
+			return 0, nil
+		case event := <-events:
+			return event, nil
+		}
+	}
```

Comment:
It feels weird to use `testEnd` like this. This isn't called in another goroutine, so what are we trying to exit?

If we want to keep an implementation that supports cancelation, I feel like we should use the passed in context instead.

Reply:
Fixed [here](https://github.com/ava-labs/avalanchego/pull/3999/commits/aa40197de464a9646a64b79fa21ff08b7d3b8db7#diff-0a19ad2bab665d2f2f629cb3a692b258012a9416ebe786715959abfd915a575dR813).